### PR TITLE
Added comments to filter.js

### LINKED ISSUE #6533: Add Comments to "filters.js" for Better Code Understanding.
### Increasing Access

We have noticed that the [`src/image/filters.js`](https://github.com/processing/p5.js/blob/main/src/image/filters.js) file contains code for various filters, but it's quite concise, making it a bit challenging to understand (For eg : Blur implementation is really optimized, which unfortunately also makes it pretty hard to read). We would like to add comments to the code, which will help both myself and other contributors comprehend the code more easily. Comments will be added to the "filters.js" file, making it more understandable and encouraging other contributors to contribute to it.

### Most appropriate sub-area of p5.js?

- [ ] Accessibility
- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] DOM
- [ ] Events
- [X] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [X] WebGL
- [ ] Build Process
- [ ] Unit Testing
- [ ] Internalization
- [ ] Friendly Errors
- [ ] Other (specify if possible)

### Feature enhancement details

If there are complex logic or algorithms involved, provide a step-by-step explanation as well.

### PR DESCRIPTION
**Pull request by perminder-17**
Resolves #6533 

 Changes:

 Screenshots of the change:

#### PR Checklist

- [x] `npm run lint` passes
- [x] [Inline documentation] is included / updated
- [x] [Unit tests] are included / updated

[Inline documentation]: https://github.com/processing/p5.js/blob/main/contributor_docs/inline_documentation.md
[Unit tests]: https://github.com/processing/p5.js/tree/main/contributor_docs#unit-tests

### CODE REVIEW

**Comment by perminder-17** [comment]
When we use filter(posterize, false), the expected behavior is to apply posterization at level 4, which is the default value. However, currently, it defaults to a level of 0. In the pull request, I've fixed this issue too. Should I create a new pull request specifically for this since it's a small matter?

File: src/image/filters.js [hunk 1]
```diff
@@ -195,29 +205,31 @@ const Filters = {
...
+ * @param  {Canvas} canvas Canvas to apply thershold filter on.
+ * @param  {Float} level Threshold level (0-1).
  */
   threshold(canvas, level) {
     const pixels = Filters._toPixels(canvas);
 
     if (level === undefined) {
       level = 0.5;
     }
+    // Calculate thershold value on a (0-255) scale.
```
Comment (davepagurek) [inline src/image/filters.js:217]:
```suggestion
    // Calculate threshold value on a (0-255) scale.
```

File: src/image/filters.js [hunk 2]
```diff
@@ -288,7 +300,9 @@ const Filters = {
  */
   posterize(canvas, level) {
     const pixels = Filters._toPixels(canvas);
-
+    if (level === undefined) {
+      level = 4;
```
Comment (davepagurek) [inline src/image/filters.js:304]: Good catch!

**Review by davepagurek — COMMENTED** [review]
Found one typo, otherwise looks good!

**Review by davepagurek — APPROVED** [review]

**Comment by perminder-17** [comment]
@davepagurek Thankyou

**Pull request merged**